sklearn/utils/__init__.py

```diff
@@ -187,19 +188,18 @@ def safe_indexing(X, indices, axis=0):
     X : array-like, sparse-matrix, list, pandas.DataFrame, pandas.Series
         Data from which to sample rows, items or columns.
     indices : array-like
-        - When ``axis=0``, indices need to be an array of integer.
-        - When ``axis=1``, indices can be one of:
+        - For both `axis=0` and `axis=1`, indices can be one of:
             - scalar: output is 1D, unless `X` is sparse.
               Supported data types for scalars:
                 - integer: supported for arrays, sparse matrices and
                   dataframes.
-                - string (key-based): only supported for dataframes.
             - container: lists, slices, boolean masks: output is 2D.
               Supported data types for containers:
-                - integer or boolean (positional): supported for
-                  arrays, sparse matrices and dataframes
-                - string (key-based): only supported for dataframes. No keys
-                  other than strings are allowed.
+                - integer or boolean (positional): supported for arrays, sparse
+                  and dataframes
+        - If `X` is a dataframe and `axis=1`, indices support string data type
+          (key-based) as a scalar or a container. The output dimension will be
```

Comment:
how would you reword it though?

Reply:
Uhm not sure what's best here.

Reply:
If I understand correctly, I'd propose:

> If `X` is a dataframe and `axis=1`, `indices` can be a string or a list/array of strings. Output is 2D.

I wouldn't use the term container here since it says above a container can be a slice or a boolean mask, which makes no sense for strings.